src/Simplify.cpp

```diff
@@ -3908,6 +3924,57 @@ class Simplify : public IRMutator {
             IRMutator::visit(op);
         }
     }
+    template <typename T>
+    Expr hoist_slice_vector(Expr a, Expr b) {
+        debug(4) << "Trying to hoist slice vector:\n... a = " << a << "\n... b = " << b << "\n";
+
+        const Call *call_a = a.as<Call>();
+        const Call *call_b = b.as<Call>();
+
+        internal_assert(call_a && call_b &&
+                        call_a->is_intrinsic(Call::slice_vector) &&
+                        call_b->is_intrinsic(Call::slice_vector));
+
+        if (!equal(call_a->args[1], call_b->args[1]) ||
+            !equal(call_a->args[2], call_b->args[2])) {
+            debug(4) << " ...unable to hoist - slice vector arguments don't match\n";
+            return T::make(a, b);
```

Comment:
In the case where this branch is taken (or any other that returns `T::make(a, b)`), this logic re-builds an identical expression to the original. We generally try to avoid that (the same_as logic in all of the IRMutators). @abadams, any opinion on the impact of this?